reference/activity_manager/project_completeness.rst

```diff
@@ -0,0 +1,102 @@
+What is the project completeness?
+=================================
+
+Goal
+____
+
+The Activity Manager feature is about creating project from a products selection that give vision to contributors (users
+in charge of enrichment). Each project gives to the contributor its own completeness on the products selection, taking
+in account its permissions to edit or not products, attribute groups, locales, scopes.
+
+We need to go deeper
+____________________
+
+The need is to apply permissions of each user on a products selection in order to acquire a new sub-selection. Then
+calculate which percentage of this sub-selection is not started to enrich, in progress or done. That is to say for each
+user and for each product in the selection apply permissions and in the user perimeter watching if there is at least one
+attribute filled or all attributes filled.
+
+Doing this synchronously is a huge task and absolutely not possible in the present state of PIM.
+
+One table to rule them all
+__________________________
+
+We found two ways to deal with this problem. First one was to calculate this detailed completeness for each user and
+cache it in a table. Benefit of this method is clear, each time we need those numbers we simply query the database: "Hey
+you, I just met Mary, what is her completeness? TODO: 42, IN_PROGRESS: 1251, DONE: 3254.". And this is crazy, result is
+instant, but the drawback, among others, is that the data is not up to date as this table is a cache. It needs to be
+calculate as often as possible but the task is so huge that with many different projects and users a night could not
+be sufficient.
+
+This first method seems not to be the chosen one. Let's recap which data we need:
```

Comment:
Well IMHO you should not write about what idea has been abandoned, it's useless as reader/integrator, I want to know how it's done/implemented. So maybe you should drop the whole block above.